test/unit/controllers/todo.controller.test.ts

```diff
@@ -0,0 +1,130 @@
+import { expect } from '@loopback/testlab';
+import { TodoController } from '../../../src/controllers';
+import { TodoRepository } from '../../../src/repositories';
+import * as sinon from 'sinon';
+import { Todo } from '../../../src/models/index';
+
+describe('TodoController', () => {
+  let todoRepo: TodoRepository;
+
+  // Method stubs
+  let create: sinon.SinonStub;
+  let findById: sinon.SinonStub;
+  let find: sinon.SinonStub;
+  let replaceById: sinon.SinonStub;
+  let updateById: sinon.SinonStub;
+  let deleteById: sinon.SinonStub;
+
+  // Test fixtures
+  let controller: TodoController;
+  let todo: Todo;
+  let todoWithId: Todo;
+  let changedTodo: Todo;
+  let todoList: Todo[];
+
+  const noError = 'No error was thrown!';
+
+  describe('createTodo', () => {
+    beforeEach(resetRepositories);
+    it('creates a Todo', async () => {
+      expect(await controller.createTodo(todo)).to.eql(todoWithId);
```

Comment:
When reading this test, I find it confusing why `createTodo` is supposed to return `todoWithId`. It would be better to setup the behaviour of the relevant stub here in this test, even if it means a bit of duplication. I do realise I recommended to move as much of the shared/repeated setup to a "beforeEach" hook before. The difficult part is to strike the right balance between making the test readable by including all relevant bits and not repeating too much stuff.

I also find this style, where the "act" part is on the same line as the first "assert" part, a bit difficult to quickly comprehend.

Here is how I would write this test myself (minus the comments). The name "aTodoWithId" (prefixed with "a") is hinting that any Todo instance can be used here, that there is nothing special about the property values used. It's a convention I found useful in the past, but we don't necessarily have to apply it in LoopBack too.

```ts
it('creates a Todo', async () => {
  // arrange
  create.resolves(aTodoWithId);
  // act
  const result = await controller.createTodo(todo);
  // assert
  expect(result).to.eql(aTodoWithId);
  sinon.assert.calledWith(create, todo);
});
```

This comment applies to other tests too.